system/Security/Security.php

```diff
@@ -163,35 +176,22 @@ class Security implements SecurityInterface
      */
     private ?string $hashInCookie = null;
 
+    /**
+     * Security Config
+     */
+    protected SecurityConfig $config;
+
     /**
      * Constructor.
      *
      * Stores our configuration and fires off the init() method to setup
      * initial state.
      */
-    public function __construct(App $config)
+    public function __construct(SecurityConfig $config)
     {
-        $security = config(SecurityConfig::class);
-
-        // Store CSRF-related configurations
-        if ($security instanceof SecurityConfig) {
-            $this->csrfProtection = $security->csrfProtection ?? $this->csrfProtection;
-            $this->tokenName      = $security->tokenName ?? $this->tokenName;
-            $this->headerName     = $security->headerName ?? $this->headerName;
-            $this->regenerate     = $security->regenerate ?? $this->regenerate;
-            $this->redirect       = $security->redirect ?? $this->redirect;
-            $this->rawCookieName  = $security->cookieName ?? $this->rawCookieName;
-            $this->expires        = $security->expires ?? $this->expires;
-            $this->tokenRandomize = $security->tokenRandomize ?? $this->tokenRandomize;
```

Comment:
I think it was quite useful. It guarantees that there is no way that the required variables will not be set. Especially if one day we add new variables to the config.

Reply:
I got what missing config variables are.

Reply:
1. I don't want to have too many properties in a class.
2. I don't want to have copies of config items in a class.
3. I don't want to have the code above.

So I changed to having the config object in the class.
I already changed like this in `Session`:
https://github.com/codeigniter4/CodeIgniter4/blob/10d3be685a2d8ddb9995c1997ca13e24727a4ca0/system/Session/Session.php#L187

But I understand your concern about what if we add a new config item.

Reply:
It seems to me that if a Config item is added, users should change the Config file as well.

Probably it would mean that a feature has been added, and it would be a good opportunity to know about it.

Reply:
Ok. That is a fair point.